packages/slate-plugins/src/common/types.ts

```diff
@@ -22,6 +22,10 @@ export type RenderElement = (
   props: RenderElementProps
 ) => JSX.Element | undefined;
 
+export interface TElement extends Element {
+  type: string;
```

Comment:
if there's only a fixed set of types, e.g. mention, link, etc., we could leverage here [discriminated unions](https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions) for better type inference.
If the above is true we could use here `enum` listing all the possible values, or a union of string literals.

Reply:
That would be nice but the type is an option of the plugins (quite important if storing in a database) so we can't use string literal types